src/psyclone/tests/dependency_test.py

```diff
@@ -369,30 +387,13 @@ def test_location(parser):
     assert x_accesses[0].location == x_accesses[1].location
 
 
-def test_user_defined_variables(parser):
```

Comment:
Again. could you please explain why this test is removed.

Reply:
This is a sort of "two steps forward, one step back". We're making heavier use of the PSyIR which is clearly the future. However, the PSyIR does not yet support user-defined (derived) types so I've taken the decision to remove this functionality entirely for the moment. However, in writing that I realised that I should keep the test as it is functionality we will need. Therefore, I've put the test back and changed it slightly so that it still works.

Reply:
Sneaky